app/adopt_a_paw/page.tsx

```diff
@@ -9,51 +9,57 @@ import '../../styles.css';
 import { Pet } from '../../types/pet-types';
 import { faHeart } from '@fortawesome/free-solid-svg-icons';
 import { fetchPets } from '../../utils/api/api';
-import { useAppDispatch, useAppSelector } from '../../app/GlobalRedux/store';
+import { useAppDispatch, useAppSelector } from '../GlobalRedux/store';
 import { setCategoryState } from '../GlobalRedux/Feautures/category/category-slice';
+import { setPetsState } from '../GlobalRedux/Feautures/pets/pets-slice';
+import { setIsLoading } from '../GlobalRedux/Feautures/loading/loading-slice';
+import { setShowFavorites } from '../GlobalRedux/Feautures/popup/showFavorites-slice';
+import { setShowForm } from '../GlobalRedux/Feautures/popup/showForm-slice';
+import { setShowPetDetail } from '../GlobalRedux/Feautures/popup/showPetDetail-slice';
+import { setSelectedPet } from '../GlobalRedux/Feautures/pets/selectedPet-slice';
+import { setFormSelectedPet } from '../GlobalRedux/Feautures/pets/formSelectedPet-slice';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-
-interface Props {
-  checkout: () => void;
-  removeItem: () => void;
-}
+import { setFavoritesItems } from '../GlobalRedux/Feautures/favorites/favoritesItems-slice';
+import { setVisiblePets } from '../GlobalRedux/Feautures/pets/visiblePets-slice';
+import { setIsFavoritesEmpty } from '../GlobalRedux/Feautures/favorites/isFavoritesEmpty-slice';
+import { setLastId } from '../GlobalRedux/Feautures/favorites/lastId-slice';
 
 const AdoptAPaw = () => {
-  // Pets state
-  const [pets, setPets] = useState<Pet[]>([]);
-  const [isLoading, setLoading] = useState<boolean>(true);
-
-  // Category
   const dispatch = useAppDispatch();
-  const categoryState = useAppSelector((state) => state.category.categoryState);
-
-  // Pet detail state
-  const [showPetDetail, setShowPetDetail] = useState<boolean>(false);
-  const [selectedPet, setSelectedPet] = useState<Pet | null>(null);
-
-  // Favorites state
-  const [favoritesItems, setFavoritesItems] = useState<Pet[]>([]);
-  const [isFavoritesEmpty, setIsFavoritesEmpty] = useState<boolean>(true);
-  const [lastId, setLastId] = useState<number>(0);
-  const [showFavorites, setShowFavorites] = useState<boolean>(false);
 
-  // Form state
-  const [formSelectedPet, setFormSelectedPet] = useState<Pet | null>(null);
-  const [showForm, setShowForm] = useState<boolean>(false);
-
-  // Show more state
-  const [visiblePets, setVisiblePets] = useState<number>(12);
+  const pets = useAppSelector((state) => state.pets.pets);
+  const selectedPet = useAppSelector((state) => state.selectedPet.selectedPet);
+  const formSelectedPet = useAppSelector(
+    (state) => state.formSelectedPet.formSelectedPet
+  );
+  const category = useAppSelector((state) => state.category.category);
+  const isLoading = useAppSelector((state) => state.isLoading.isLoading);
+  const showFavorites = useAppSelector(
```

Comment:
Redux has combineSelectors for combines them all to redce repeating code

Reply:
Done